### DISCUSSION TITLE
Windows Package Manager 1.5.1081-preview

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#3181] [Releases]
This is the third development build after the Windows Package Manager 1.4 build for Windows 10 (1809+) and Windows 11. This build will be released to Windows Insider Dev builds and Windows Package Manager Insiders.

Experimental features are enabled in this release. The experimental feature for enabling Windows Features is now supported and included in this release. You can now specify the Windows feature name (i.e. **netfx3**) you want to enable in the [WindowsFeature field](https://github.com/microsoft/winget-cli/blob/a46ef6cab7a33b7d446484ac9eaa02028d723744/schemas/JSON/manifests/v1.4.0/manifest.installer.1.4.0.json#L297-L307) in your installer manifest.

Run `winget features` to see which experimental features are enabled or disabled.
Add the following to your settings (`winget settings`) file to enable the experimental features including package pinning:

```JSON
    "experimentalFeatures": {
	  "pinning": true,
	  "dependencies": true,
	  "directMSI": true,
	  "uninstallPreviousArgument": true,
	  "windowsFeature": true,
    },
```

This release makes improvements to the overall user experience of WinGet with improvements to package correlation, COM APIs, logging/output, and Portable installation. This release also includes an early preview of our Microsoft.WinGet.Client PowerShell module. Information about getting started and usage can be found [here](https://github.com/microsoft/winget-cli/tree/master/src/PowerShell/Microsoft.WinGet.Client#quick-start-guide).

## What's Changed
* Add std::endl after enabling/disabling Admin Setting by @Trenly in https://github.com/microsoft/winget-cli/pull/2958
* Update settings.schema.0.2.json to include pinning feature by @florelis in https://github.com/microsoft/winget-cli/pull/2961
* Update expect.txt by @RDMacLachlan in https://github.com/microsoft/winget-cli/pull/2968
* Config code by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/2942
* Add alternate url support for some predefined sources by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/2970
* Remove zip from 1.2 schema by @Trenly in https://github.com/microsoft/winget-cli/pull/2996
* Remove zip installer type from 1.0 and 1.1 manifests by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3006
* Fix order of pin labels by @Trenly in https://github.com/microsoft/winget-cli/pull/3016
* Configuration processors by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3008
* docs: fix "default install root" location by @Lutra-Fs in https://github.com/microsoft/winget-cli/pull/3034
* Microsoft.Management.Configuration Windows target platform by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3032
* Fix typo in MSStore get entitlement log by @yao-msft in https://github.com/microsoft/winget-cli/pull/3031
* AnyCPU for Microsoft.Management.Configuration.Projection and Microsoft.Management.Configuration.Processor by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3041
* Expose Agreements and Locale Fields in COM by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/2897
* Configuration commands by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3036
* User/romaclac/spec enable store UI bypass by @RDMacLachlan in https://github.com/microsoft/winget-cli/pull/3035
* Clarify `No installed package matching the input criteria was found` in upgrade flow by @Trenly in https://github.com/microsoft/winget-cli/pull/2877
* Use copy instead of rename for moving extracted archive directories by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3003
* Add installed files and startup link metadata collection in Installer Metadata collection util by @yao-msft in https://github.com/microsoft/winget-cli/pull/2985
* Explicitly prevent fallthrough on OperationType by @Trenly in https://github.com/microsoft/winget-cli/pull/3075
* Schema draft for WinGet configure by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/2851
* Fix PATH behavior of non-symlink installations for Portables/Zip by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3002
* Define properties object in configuration schema by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3081
* Make sure source agreements are initialized before checking by @yao-msft in https://github.com/microsoft/winget-cli/pull/3080
* Rework extra packaged files mechanism by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3064
* Add new icon assets by @denelon in https://github.com/microsoft/winget-cli/pull/3086
* Add diagnostics to the processor component by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3087
* Implement PowerShell cmdlets by @msftrubengu in https://github.com/microsoft/winget-cli/pull/2838
* Add logging around specific number of matches by @Trenly in https://github.com/microsoft/winget-cli/pull/3094
* Reformat Microsoft.WinGet.Client module output by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3088
* Make ConfigurationRemotingServer selfcontained by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3097
* Processor improvements by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3101
* Put required modules into the repo and consume them in dev scenarios by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3103
* Throw exception if Import-Module fails by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3104
* Add experimental feature support for enabling Windows Feature dependencies by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3005
* Fix internal build by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3106
* Update the privacy statement link in README to link to correct page by @ShyAssassin in https://github.com/microsoft/winget-cli/pull/3116
* Remove one unnecessary lock per localization team suggestion by @yao-msft in https://github.com/microsoft/winget-cli/pull/3115
* Pass COM caller name to rest source in request header by @yao-msft in https://github.com/microsoft/winget-cli/pull/3112
* Enforce single install across winget processes by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3118
* Support to Bypass Store Client App Policy When Called Through COM by @PaulJYim in https://github.com/microsoft/winget-cli/pull/3105
* Bug fixes by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3127
* Use in-proc winget in PowerShell module when running as SYSTEM by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3129
* Improve correlation by keeping arch info declared in manifest arp DisplayName entry by @yao-msft in https://github.com/microsoft/winget-cli/pull/3100
* Show Portable Installation Directories with --info by @Trenly in https://github.com/microsoft/winget-cli/pull/3128
* Use FolderId_ProgramFiles for non x86 default machine portable folder by @yao-msft in https://github.com/microsoft/winget-cli/pull/3137
* Add Windows10SDK 19041 Workload Component to VSConfig File by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3143
* Add support for bool, strings and arrays in Configuration settings by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3135
* Add isPublic to IConfigurationUnitProcessorDetails by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3145
* fix PRIVACY.md links by @ave9858 in https://github.com/microsoft/winget-cli/pull/3147
* Update README.md by @mdanish-kh in https://github.com/microsoft/winget-cli/pull/3148
* Add refreshed Store source certs by @yao-msft in https://github.com/microsoft/winget-cli/pull/3149
* Add configuration telemetry events by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3152
* Improve configure user experience by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3158
* Fix bugs with display paths by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3157

## New Contributors
* @RDMacLachlan made their first contribution in https://github.com/microsoft/winget-cli/pull/2968
* @Lutra-Fs made their first contribution in https://github.com/microsoft/winget-cli/pull/3034
* @ShyAssassin made their first contribution in https://github.com/microsoft/winget-cli/pull/3116
* @PaulJYim made their first contribution in https://github.com/microsoft/winget-cli/pull/3105
* @ave9858 made their first contribution in https://github.com/microsoft/winget-cli/pull/3147
* @mdanish-kh made their first contribution in https://github.com/microsoft/winget-cli/pull/3148

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/v1.5.441-preview...v1.5.1081-preview

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.5.1081-preview'>Windows Package Manager 1.5.1081-preview</a>.</em>

**Comment by manuel-rojas-a** [comment]
To install type from PowerShell

Add-AppxPackage -Path https://github.com/microsoft/winget-cli/releases/download/v1.5.1081-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
